### PR TITLE
=act #19466 add failing test for `ByteString`

### PR DESCRIPTION
**Pull request by briantopping**
This obviously needs a fix before it can be committed. Please also see romix/akka-kryo-serialization#69 for more background here and a workaround that @luben very kindly solved and put together. :+1:

### CODE REVIEW

**Comment by johanandren** [comment]
Refs #19466

File: akka-actor-tests/src/test/scala/akka/util/ByteStringSpec.scala [hunk 1]
```diff
@@ -617,5 +618,10 @@ class ByteStringSpec extends WordSpec with Matchers with Checkers {
         }
       }
     }
+    "return the correct class from Builder.result" in {
+      val builder = ByteString().genericBuilder[String]
```
Comment (viktorklang) [inline akka-actor-tests/src/test/scala/akka/util/ByteStringSpec.scala]: Is there any reason why this method should not throw an UnsupportedOperationException?

Use ByteString.newBuilder instead?

**Comment by ktoso** [comment]
For reference, the origin of this issue is resolved:
The original problem case was kryo serialization but it now handles `ByteString` as first class citizen thanks to @luben's fix here: https://github.com/romix/akka-kryo-serialization/pull/70

**Comment by briantopping** [comment]
Did Kryo work around it or is it a base implementation problem?

**Comment by viktorklang** [comment]
@briantopping What should a `genericBuilder` for ByteString look like?

**Comment by briantopping** [comment]
I see.. thanks!

**Comment by luben** [comment]
Just adding a context for the history:

The akka-kryo-serialization has a generic serializer/deserializer for the collections that implement Traversable[T] trait. This serializer gets picked for ByteString because it implements the trait through IndexedSeq[T]. On deserializing the object it uses the genericBuilder method and in the case with ByteString it get a VectorBuilder instance because this is inherited from the immutable.IndexedSeq.

So I don't think ByteString should raise Unsupported Operation exception - it is part if the Traversable trait that it implements. The problem was solver on the akka-kryo-serialization side by implementing specialized serializer/deserializer for ByteString - it has a better performance as well.

**Comment by ktoso** [comment]
Right, and thanks a lot for the fix in the lib and chiming in here @luben!

**Pull request closed without merging**